app/src/main/java/edu/mills/heartoftaiwanese/activity/home/HomeFragment.kt

```diff
@@ -0,0 +1,213 @@
+package edu.mills.heartoftaiwanese.activity.home
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import com.google.android.material.tabs.TabLayout
+import edu.mills.heartoftaiwanese.R
+import edu.mills.heartoftaiwanese.activity.BaseFragment
+import edu.mills.heartoftaiwanese.activity.hideKeyboard
+import edu.mills.heartoftaiwanese.databinding.FragmentHomeBinding
+import edu.mills.heartoftaiwanese.network.WebResultCode
+import java.util.Calendar
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [HomeFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class HomeFragment : HomeContract.HomeView, BaseFragment(), TabLayout.OnTabSelectedListener {
+    companion object {
+        private const val kSavedChineseText = "SavedChineseText"
+        private const val kSavedEnglishText = "SavedEnglishText"
+
+        /**
+         * Use this factory method to create a new instance of
+         * this fragment using the provided parameters.
+         *
+         * @return A new instance of fragment HomeFragment.
+         */
+        @JvmStatic
+        fun newInstance() =
+            HomeFragment()
+    }
+
+    private lateinit var binding: FragmentHomeBinding
+    private lateinit var viewModel: HomeViewModel
+    private var currentTab: Int? = 0
+
+    override fun onSaveInstanceState(state: Bundle) {
+        super.onSaveInstanceState(state)
+        state.putString(kSavedChineseText, binding.editTextCh.text.toString())
+        state.putString(kSavedEnglishText, binding.editTextEng.text.toString())
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentHomeBinding.inflate(layoutInflater, container, false)
+        savedInstanceState?.let {
+            binding.editTextCh.setText(it.getString(kSavedChineseText))
+            binding.editTextEng.setText(it.getString(kSavedEnglishText))
+        }
+        val currentHour: Int = Calendar.getInstance().get(Calendar.HOUR_OF_DAY)
+        binding.tvTranslateExplanation.text = getString(
+            R.string.translationFragmentExplanation,
+            isMorningAfternoonEvening(currentHour)
+        )
+        viewModel = HomeViewModel()
+        viewModel.configure(this, requireActivity().applicationContext)
+        initializeClickListeners()
+
+        binding.editTextCh.visibility = View.GONE
+        binding.submitCh.visibility = View.GONE
+        return binding.root
+    }
+
+    private fun isMorningAfternoonEvening(hour: Int): String {
+        return when (hour) {
```

Comment:
Switch to a function expression.

It would be nice to have a comment explaining these magic numbers.